File: bindings/nodejs/src/clients/productRpcClient.js
const { RpcSession } = require("../rpcSession");

class CenyslovenskoProductRpcClient {
  constructor(options = {}) {
    this.session = options.session || new RpcSession(options);
  }

  start() {
    this.session.start();
  }

  close() {
    this.session.close();
  }

  async call(method, params) {
    return this.session.call(method, params);
  }

  async getProduct(productId) {
    return this.call("product.get", { id: productId });
  }
}

module.exports = {
  CenyslovenskoProductRpcClient,
};

File: bindings/ruby/lib/cenyslovensko_bindings/clients/product_rpc_client.rb
module CenyslovenskoBindings
  module Clients
    class ProductRpcClient
      def initialize(session: nil, **session_options)
        @session = session || RpcSession.new(**session_options)
      end

      def start
        @session.start
      end

      def close
        @session.close
      end

      def call(method, params = nil)
        @session.call(method, params)
      end

      def get_product(product_id)
        call("product.get", { id: product_id })
      end
    end
  end
end
